### PR TITLE
Log Restore Config data

### PR DESCRIPTION
**Pull request by gbova**
Add logging for `RestoreConfig` and a more helpful representation of `TimingContext`.
Hopefully this will help us with https://dimagi-dev.atlassian.net/browse/HI-663

code buddy: @orangejenny

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 1]
```diff
@@ -669,6 +669,9 @@ def _get_asynchronous_payload(self):
             # recorded since it is async (see self.get_response).
             with self.timing_context("wait_for_task_to_start"):
                 task = get_async_restore_payload.delay(self, self.domain, self.restore_user.username)
+                logging.info('RestoreConfig after get_async_restore_payload task is created: {msg}'.format(
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]:
```suggestion
                logger.info('RestoreConfig after get_async_restore_payload task is created: {msg}'.format(
```

Comment (millerdev) [reply corehq/ex-submodules/casexml/apps/phone/restore.py]: nit: the logging package is pre-wired to do %-style string formatting:
```py
logger.info("message: %s", msg)
```
The cool thing about this is that string formatting is deferred until it is known that the log message will actually be written somewhere. So if the logging level is set to ERROR or WARNING (meaning INFO messages are ignored), then the string formatting would not be done for this example, which saves some time and processing resources. Unfortunately this does not work for `string.format()`-style formatting.

So you could make this a bit more efficient by changing it to
```py
logging.info('RestoreConfig after get_async_restore_payload task is created: %r', self)
```

File: corehq/ex-submodules/casexml/apps/phone/tasks.py [hunk 2]
```diff
@@ -46,12 +46,9 @@ def get_async_restore_payload(restore_config, domain=None, username=None):
     Process an async restore
     domain and username: added for displaying restore request details on flower
     """
-    try:
-        repr(restore_config)
-    except Exception as e:
-        logging.error('Something went wrong with RestoreConfig.__repr__() : {msg}'.format(
-            msg=str(e)
-        ))
+    logging.info('RestoreConfig in get_async_restore_payload task: {msg}'.format(
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/tasks.py]:
```suggestion
    log.info('RestoreConfig in get_async_restore_payload task: {msg}'.format(
```

Also somewhere near the top of the file just below the imports you'll need to add
```py
log = logging.getLogger(__name__)
```

**Review by millerdev — APPROVED** [review]
The logging changes are not absolutely necessary, but should be done so that the log messages are correctly namespaced. Could do that in a separate PR if this one cannot be updated in time for deploy.

File: corehq/ex-submodules/casexml/apps/phone/restore.py (hunk 1 above)
Comment (gbova) [reply corehq/ex-submodules/casexml/apps/phone/restore.py]: Nice. Thank you!

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 3]
```diff
@@ -63,6 +63,8 @@
 
 
 logger = logging.getLogger(__name__)
+log = logging.getLogger(__name__)
+
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: This is not necessary. In this module you can use the one that's already created on the line above (`logger`). Sorry for the naming confusion.

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 4]
```diff
@@ -760,7 +763,6 @@ def _record_timing(self, status):
             else:
                 sync_log = self.restore_state.current_sync_log
                 sync_log_id = sync_log._id if sync_log else 'N/A'
-            log = logging.getLogger(__name__)
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: I think this should not be removed. I'm forgetting why it's here, but in some contexts the global `logger` was not working in this function. I hope it doesn't turn out to be that way for your logging call above.

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 5]
```diff
@@ -572,7 +574,7 @@ def get_response(self):
                 payload = self.get_payload()
             response = payload.get_http_response()
         except RestoreException as e:
-            logging.exception("%s error during restore submitted by %s: %s" %
+            log.exception("%s error during restore submitted by %s: %s" %
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]:
```suggestion
            logger.exception("%s error during restore submitted by %s: %s" %
```

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 6]
```diff
@@ -669,6 +671,7 @@ def _get_asynchronous_payload(self):
             # recorded since it is async (see self.get_response).
             with self.timing_context("wait_for_task_to_start"):
                 task = get_async_restore_payload.delay(self, self.domain, self.restore_user.username)
+                logging.info('RestoreConfig after get_async_restore_payload task is created: %r', self)
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]:
```suggestion
                logger.info('RestoreConfig after get_async_restore_payload task is created: %r', self)
```

**Pull request closed without merging**